### ISSUE TITLE
Membrane size on addMembranes

### ISSUE THREAD

**Issue by JoaoRodrigues** [openmm/openmm#2435]
I've been trying to optimize the speed of my simulations and the one issue I have found is the size of the membrane generated by ` modeller.addMembrane`. The docstring of the method says the following:

>The size of the membrane and water box are determined by the minimumPadding argument. All pre-existing atoms are guaranteed to be at least this far from any edge of the periodic box.  It is also possible for the periodic box to have more padding than requested.  In particular, it only adds whole copies of the pre-equilibrated membrane patch, so the box dimensions will always be integer multiples of the patch size.  That may lead to a larger membrane than what you requested.

Is there a reason why there is no trimming done after tiling the patches? One argument I could think of against this is that the packing of the lipids between periodic images will be disrupted and will need equilibration, but then again, so does the entire system once the protein is embedded. I wrote a small script that trims membranes in the XY plan to fit the requested padding, and the simulations seem to behave OK - while reducing the number of atoms considerably (eg. 200k vs 80k). Seems like it could be a worthwhile feature to have, maybe as an option to `addMembrane`?

**Comment by peastman** [comment]
That's pretty much the reason.  Trimming lipids is a lot more complicated than removing waters.  They're much larger and more irregularly shaped, and they're much more tangled up and less neatly packed.  And you don't want to leave a big gap in the membrane, or water will flow into it and things will get really messed up.

The safest way of trimming membranes might be to do it in a series of steps.  Remove just a few lipids and run some constant pressure dynamics to let the box size equilibrate.  Then remove a few more, and keep repeating until you get down to the size you want.  Of course, each time you remove lipids you also need to remove a bit of water.

I'll mark this as an enhancement request in case someone wants to try implementing it and see if they can come up with something that works robustly.

**Comment by JoaoRodrigues** [comment]
Makes sense. What I had was "conservative", in the sense that a molecule needed to have at least 70% of its atoms outside my "radius" to be cut from the membrane. This way I did not leave gaps and never saw waters coming into the membrane. The water is indeed the big issues here in terms of number of molecules, but cutting away waters isn't a big deal.

Adding several equilibration (NPT) steps might be too expensive however. What I could think of is to calculate the size of the final box and if it's within a certain threshold of the requested padding (say, +-0.5nm) keep it, otherwise trim it and issue a warning saying the box might need a longer equilibration. Would this be an acceptable compromise?

I'll give it a shot and report back.

**Comment by peastman** [comment]
Sounds good.  We'll just need to test it thoroughly to make sure it's robust.

**Comment by peastman** [comment]
@JoaoRodrigues just wondering if you ever made any progress with this?

**Comment by JoaoRodrigues** [comment]
Hi @peastman,

I did, I'll have to find the branch where I have this working - completely forgot to get back to you about it. I'll give you back an answer in a couple of days. Thanks for the ping!

**Comment by peastman** [comment]
> I'll give you back an answer in a couple of days. Thanks for the ping!

It's been more than a couple of days. :)
